src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -2374,9 +2374,8 @@ public static void Reverse(ref byte buf, nuint length)
                 buf = ref Unsafe.Add(ref buf, numIters * numElements);
                 length -= numIters * numElements * 2;
             }
-            else if (Ssse3.IsSupported && (nuint)Vector128<byte>.Count * 2 <= length)
+            else if ((Ssse3.IsSupported || AdvSimd.Arm64.IsSupported) && (nuint)Vector128<byte>.Count * 2 <= length)
```

Comment:
`Vector128.IsHardwareAccelerated` is a more generic way of checking that:

```suggestion
            else if (Vector128.IsHardwareAccelerated && (nuint)Vector128<byte>.Count * 2 <= length)
```